proto/aserto/directory/reader/v2/reader.proto

```diff
@@ -4,36 +4,366 @@ package aserto.directory.reader.v2;
+    };
+
+    rpc GetRelationTypes(GetRelationTypesRequest) returns (GetRelationTypesResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/relation_types"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get relation types"
+            description: "Returns relation types, optionally by object_type."
+            operation_id: "directory.relation_types.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };        
+    };
     
     // permission metadata methods
-    rpc GetPermission(GetPermissionRequest) returns (GetPermissionResponse) {};
-    rpc GetPermissions(GetPermissionsRequest) returns (GetPermissionsResponse) {};
+    rpc GetPermission(GetPermissionRequest) returns (GetPermissionResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/permission"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get permission"
+            description: "Returns permission by name."
+            operation_id: "directory.permission.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };    
+    };
+
+    rpc GetPermissions(GetPermissionsRequest) returns (GetPermissionsResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/permissions"
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+            tags: "directory"
+            summary: "Get permissions"
+            description: "Returns permissions."
+            operation_id: "directory.permissions.get"
+            deprecated: false
+            security: {
+                security_requirement: {
+                    key: "TenantID";
+                    value: {}
+                }
+                security_requirement: {
+                    key: "DirectoryAPIKey";
+                    value: {}
+                }
+            }
+        };    
+    };
     
     // object methods
-    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {};
+    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/object"
```

Comment:
I know that some Get requests have complex structure with optional inputs and need to be exposed as POST endpoints, but for simple ones like this, would it make sense to expose it as: 
`get: "/api/v2/directory/objects/{type}/{key}`

And similarly in the writer we'd have `put: "/api/v2/directory/objects/{type}/{key}` and the same for delete.